pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -367,9 +367,9 @@ func propagateParams(t v1.PipelineTask, stringReplacements map[string]string, ar
 				}
 			}
 		}
-		t.TaskSpec.TaskSpec = *resources.ApplyReplacements(&t.TaskSpec.TaskSpec, stringReplacementsDup, arrayReplacementsDup)
+		t.TaskSpec.TaskSpec = *resources.ApplyReplacements(&t.TaskSpec.TaskSpec, stringReplacementsDup, arrayReplacementsDup, objectReplacementsDup)
```

Comment:
is this implementing propagation of object parameters? if so, i think this should be in a separate PR

Reply:
Yes and no. 

For an inlined-step this is not needed at all since inside the Task, we cannot really pass complete objects. Users need to declare explicit keys of the object they want to use. 

e.g. This is allowed
```
script: |
  echo $(params.objectparam.key)
```

but this is not:
```
script: |
  echo $(params.objectparam[*])
```
For this reason, we never passed object params for replacements inside a Task.

However now, with `StepActions`, a `Step` will need to pass it object param completely. 
For example, this is allowed:
```
steps:
  - ref:
      name: step-action
  params:
    - name: object-param
       value: $(params.objectparam[*])
```
Therefore, we now need to allow passing of objectParams into a Task. 
If you look at the [usage](https://github.com/tektoncd/pipeline/pull/7332/files#diff-487eb42be2c62df773ee4e76e5570d045aeac1ac0c7d56e095ebf14a7deac58bR374-R376), we only limit it to the use for `params` in `Steps`, and nowhere else so that the existing behaviour for inlined step remains intact.

Reply:
Makes sense, thanks for the detailed explanation. Please add this note in the commit message so that we have a record of why this change is included 🙏🏾

Reply:
Done! I added it to changes section.